### PR TITLE
[FLINK-17761][connector/common] Add a constructor taking capacity as a parameter for `FutureCompletingBlockingQueue`

### PR DESCRIPTION
**Pull request by pyscala**
## What is the purpose of the change
Add a constructor taking capacity as a parameter for `FutureCompletingBlockingQueue`

## Brief change log
Add a constructor taking capacity as a parameter for `FutureCompletingBlockingQueue`
`private final static Integer DEFAULT_CAPACITY = 10000;

public FutureCompletingBlockingQueue(FutureNotifier futureNotifier) {
   this(futureNotifier, DEFAULT_CAPACITY);
}

public FutureCompletingBlockingQueue(FutureNotifier futureNotifier, int capacity) {
   super(capacity);
   this.futureNotifier = futureNotifier;
}`

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (no), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by pyscala** [comment]
Hi @becketqin ,appreciate for your time to review this PR,thanks

**Comment by pyscala** [comment]
@flinkbot run travis

**Comment by becketqin** [comment]
@pyscala Thanks for the patch. LGTM. Can we add a unit test for this?

**Comment by pyscala** [comment]
> @pyscala Thanks for the patch. LGTM. Can we add a unit test for this?

thanks @becketqin  for your reply ,i will add a unit test for this

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.apache.flink.connector.base.source.reader.synchronization;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (becketqin) [inline flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java:5]: Can you add a license header to the file? Also, we usually do not use `*` in the imports.

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * The unit test for {@link FutureCompletingBlockingQueue}
+ */
+public class FutureCompletingBlockingQueueTest {
+
+	private static final Integer DEFAULT_CAPACITY = 10000;
```
Comment (becketqin) [inline flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java:12]: Should the unit test use a value different from the default value?

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java (hunk 1 above)
Comment (pyscala) [reply flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java:5]: @becketqin Thanks for your detail guidance . I will fix it

File: flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java (hunk 2 above)
Comment (pyscala) [reply flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/synchronization/FutureCompletingBlockingQueueTest.java:12]: Ok

**Comment by pyscala** [comment]
@flinkbot run azure

**Comment by pyscala** [comment]
@becketqin The problems have been solved, and the current error seems to have nothing to do with our changes .
Looking forward to your reply  , thanks

**Comment by pyscala** [comment]
@JingsongLi @wuchong  Can you review this PR for me? thanks.

**Comment by JingsongLi** [comment]
Thanks for the contribution. Why default value is 10000? Why not keep Int.Max?

**Comment by pyscala** [comment]
> Thanks for the contribution. Why default value is 10000? Why not keep Int.Max?

@JingsongLi Thanks for your reply , The details link  https://issues.apache.org/jira/browse/FLINK-17761 .
FutureCompletingBlockingQueue is a LinkedBlockingQueue and it is created using the default constructor, so its depth is 
Integer.MAX_VALUE. So for a FetchTask to be blocked on a put(), it means that the queue is full so it has 
Integer.MAX_VALUE elements already . Integer.MAX_VALUE is too large . I changed the default value to 10000 with @becketqin agree. At the same time, a construction method with capacity parameter is added.

**Comment by JingsongLi** [comment]
Thanks for the explanation. Looks good to me.

**Pull request merged**